electrolytes/solvation_shell_utils.py

```diff
@@ -0,0 +1,152 @@
+import re
+import numpy as np
+import random
+from typing import List, Set, Callable
+
+from schrodinger.structutils import analyze
+from schrodinger.application.matsci import clusterstruct
+from schrodinger.comparison import are_conformers
+from schrodinger.application.jaguar.utils import group_with_comparison
+from schrodinger.comparison.atom_mapper import ConnectivityAtomMapper
+from schrodinger.structutils import rmsd
+from schrodinger.structure import Structure
+
+
+def filter_shells_with_solute_atoms(
+    shells: List[Set[int]], st: Structure, solute_res_names: List[str]
+) -> List[Structure]:
+    """
+    Filter out shells that contain solute atoms.
```

Comment:
Excluding ALL solutes seems:

- a bit restrictive (probably there will be very few shells in the higher-concentration electrolytes that have exactly 0 solute molecules)
- like it could lead to over-sampling of a small number of molecules/interactions (keeping in mind that we're not actually including that many unique solvents)

I know we do want solvent-centered clusters, so maybe this is just the cleanest solution. The alternative seems like it'd be messy (how many solutes are too many? How close can they get to the central solvent? etc. etc.).

Reply:
In response to your first point, I think the intention is to only do the solvent-centered shells for low concentration systems. I've added an argument that controls whether or not we do the solvent-solvent interactions.

Regarding your other concern, I agree that it's not ideal. But maybe we handle the over-sampling problem by just sampling fewer solvent-solvent shells (i.e reduce `top_n`)?